spinn_front_end_common/interface/interface_functions/insert_edges_to_live_packet_gatherers.py

```diff
@@ -23,192 +23,177 @@
-        :param m_lpgs:\
-            dict of chip placed on to gatherers that are associated with the\
-            parameters
-        :param machine: the SpiNNaker machine object
-        :param placements: the placements object
-        :param machine_graph: the machine graph object
-        :param partition_id: the partition ID to add to the edge
-        :return: machine edge and the LPG vertex
+            for lpg_params in progress.over(live_packet_gatherer_parameters):
+                # locate vertices to connect to a LPG with these params
+                for vertex, p_ids in live_packet_gatherer_parameters[
+                        lpg_params]:
+                    self._connect_lpg_vertex_in_application_graph(
+                        application_graph, vertex, lpg_params, p_ids,
+                        n_keys_map)
+
+    def _connect_lpg_vertex_in_application_graph(
+            self, app_graph, app_vertex, lpg_params, p_ids, n_keys_map):
+        """
+        :param ~.ApplicationGraph app_graph:
+        :param ~.ApplicationVertex app_vertex:
+        :param LivePacketGatherParameters lpg_params:
+        :param list(str) p_ids:
+        :param ~.DictBasedMachinePartitionNKeysMap n_keys_map:
         """
-        # pylint: disable=too-many-arguments
-
-        # locate the LPG that's closest to this vertex
-        machine_lpg = self._find_closest_live_packet_gatherer(
-            machine_vertex, m_lpgs, machine, placements)
-
-        # add edge for the machine graph
-        machine_edges = dict()
-        for partition_id in partition_ids:
-            machine_edge = MachineEdge(machine_vertex, machine_lpg)
-            machine_graph.add_edge(machine_edge, partition_id)
-            machine_edges[partition_id] = machine_edge
-
-        # return the machine edge
-        return machine_edges, machine_lpg
-
-    @staticmethod
-    def _update_app_graph_and_mapper(
-            application_graph, graph_mapper, machine_lpg, vertex,
-            partition_ids, machine_edges, app_graph_edges, n_keys_map,
-            m_graph):
-        """ Handles changes to the application graph and graph mapper.
-
-        :param application_graph: the application graph
-        :param graph_mapper: the graph mapper
-        :param machine_lpg: the machine LPG
-        :param vertex: the application vertex to link to
-        :param machine_edges: the machine edges
-        :param m_graph: the machine graph
-        :param partition_ids: the partition IDs to put the edge on
-        :param n_keys_map: map between partition and n keys
-        :return: the application edge for this vertex and LPG
-        :rtype: ~pacman.model.graph.application.ApplicationEdge
+        if not app_vertex.machine_vertices or not p_ids:
+            return
+
+        # flag for ensuring we don't add the edge to the app
+        # graph many times
+        app_edges = dict()
+        m_graph = app_graph.machine_graph
```

Comment:
if you instead hand over the machine graph as a param to this method. this call becomes compeltely redundant